ci-operator/step-registry/cert-manager/custom-apiserver-cert/cert-manager-custom-apiserver-cert-commands.sh

```diff
@@ -0,0 +1,223 @@
+while :;
+do
+    echo "Checking the LoadBalancer service's status for the #${COUNTER}-th time ..."
+    EXTERNAL_IP_OUTPUT=$(oc get service cert-manager-managed-alt-apiserver -n openshift-kube-apiserver -o jsonpath='{.status.loadBalancer.ingress}')
+    if grep -q '"ip"' <<< "$EXTERNAL_IP_OUTPUT"; then
+        EXTERNAL_IP=$(oc get service cert-manager-managed-alt-apiserver -n openshift-kube-apiserver -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+        RECORD_TYPE=A
+        break
+    elif grep -q '"hostname"' <<< "$EXTERNAL_IP_OUTPUT"; then
+        EXTERNAL_IP=$(oc get service cert-manager-managed-alt-apiserver -n openshift-kube-apiserver -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+        RECORD_TYPE=CNAME
+        break
+    fi
+    ((++COUNTER))
+    if [[ $COUNTER -eq $MAX_RETRY ]]; then
+        echo "The LoadBalancer service's status does not show either ip or hostname after $((MAX_RETRY * INTERVAL)) seconds. Dumping status:"
+        oc get service cert-manager-managed-alt-apiserver -n openshift-kube-apiserver -o jsonpath='{.status}'
+        exit 1
+    fi
+    sleep $INTERVAL
+done
+
+BASE_DOMAIN=$(oc get dns cluster -o=jsonpath='{.spec.baseDomain}')
+ORIGINAL_API_FQDN=$(oc whoami --show-server | sed -e 's|https://||' -e 's/:6443//')
+NEW_API_FQDN=alt-api.${BASE_DOMAIN}
+
+oc create -f - << EOF
+apiVersion: ingress.operator.openshift.io/v1
+kind: DNSRecord
+metadata:
+  name: cert-manager-managed-alt-apiserver
+  namespace: openshift-ingress-operator
+spec:
+  dnsManagementPolicy: Managed
+  dnsName: "${NEW_API_FQDN}."
+  recordTTL: 30
+  recordType: ${RECORD_TYPE}
+  targets:
+  - ${EXTERNAL_IP}
+EOF
+
+# Wait for the dnsrecord status to become ready
+MAX_RETRY=12
+INTERVAL=10
+COUNTER=0
+while :;
+do
+    echo "Checking the cert-manager-managed-alt-apiserver dnsrecord status for the #${COUNTER}-th time ..."
+    DNSRECORD_STATUS="$(oc get dnsrecord cert-manager-managed-alt-apiserver -n openshift-ingress-operator '-o=jsonpath={.status.zones[*].conditions[?(@.type=="Published")].status}')"
+    if [[ ! "$DNSRECORD_STATUS" =~ False ]]; then
+        break
+    fi
+    ((++COUNTER))
+    if [[ $COUNTER -eq $MAX_RETRY ]]; then
+        echo "The cert-manager-managed-alt-apiserver dnsrecord status is still not ready after $((MAX_RETRY * INTERVAL)) seconds. Dumping status:"
+        oc get dnsrecord cert-manager-managed-alt-apiserver -n openshift-ingress-operator -o jsonpath='{.status}'
+        exit 1
+    fi
+    sleep $INTERVAL
+done
+
+CERT_NAME=alt-api-cert
+oc create -f - << EOF
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: $CERT_NAME
+  namespace: openshift-config
+spec:
+  commonName: "$NEW_API_FQDN"
+  dnsNames:
+  - "$NEW_API_FQDN"
+  usages:
+  - server auth
+  issuerRef:
+    kind: ClusterIssuer
+    name: $CLUSTERISSUER_NAME
+  secretName: cert-manager-managed-alt-api-tls
+# privateKey:
+#   rotationPolicy: Always # Venafi required this
+  duration: 2h
+  renewBefore: 1h30m
```

Comment:
These lines are intentionally needed to cover customer scenario in which how other subteams' components and cases would go when the cert would be renewed. Whether those components would fail or still succeed. A CI job runs about several hours > 2h. So these two lines are for that. If too long, no such coverage.